packages/twenty-server/src/engine/metadata-modules/object-metadata/hooks/before-update-one-object.hook.ts

```diff
@@ -11,9 +12,27 @@ import {
 } from '@ptc-org/nestjs-query-graphql';
 import { Equal, In, Repository } from 'typeorm';
 
+import { WorkspaceEntityMetadataArgs } from 'src/engine/twenty-orm/interfaces/workspace-entity-metadata-args.interface';
+
 import { FieldMetadataEntity } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
 import { UpdateObjectPayload } from 'src/engine/metadata-modules/object-metadata/dtos/update-object.input';
 import { ObjectMetadataService } from 'src/engine/metadata-modules/object-metadata/object-metadata.service';
+import { metadataArgsStorage } from 'src/engine/twenty-orm/storage/metadata-args.storage';
+import { standardObjectMetadataDefinitions } from 'src/engine/workspace-manager/workspace-sync-metadata/standard-objects';
+
+function getStandardObjectMetadata(
```

Comment:
Think about efficiency of this function

E.g.
```
const standardObjectMap = new Map<string, WorkspaceEntityMetadataArgs>();

// Populate once and keep in RAM
standardObjectMetadataDefinitions.forEach(entity => {
  const metadata = metadataArgsStorage.filterEntities(entity);
  if (metadata?.standardId) {
    standardObjectMap.set(metadata.standardId, metadata);
  }
});

//  lookup is now O(1) instead of O(n)
function getStandardObjectMetadata(standardId: string) {
  return standardObjectMap.get(standardId);
}
```

Also regardless of performance, prefer using keywords like `find` instead of loops

```
return standardObjectMetadataDefinitions.find(entity => metadataArgsStorage.filterEntities(entity)?.standardId === standardId).
```

Reply:
Modified the function according to your suggestion, moved to a specific file (also made an arrow function to match the other utilities).